risc-v-demo/Readme.md

```diff
@@ -0,0 +1,415 @@
+(bayer interpolation, gamma correction, enhancement and scaling), which are part 
+of the original AppNote. Then the video frames go though some additional 
+processing (HLS modules) before they get encoded by the H264 module.  The RISC-V
+processor then encapsulates the encoded frames into RTP packets and stream them
+over the network. The video can be watched using a video player and the entire 
+design can be controlled via web.
+
+<img src="sev-kit-reference-design/doc/overal_demo_architecture.png" width="100%">
+
+As a general concept, the HLS pipeline in the figure shows some hypothetical 
+C++ functions, shown as gray boxes, that can be compiled using SmartHLS and target 
+either the FPGA fabric or the RISC-V processors. Functions `foo()`, `bar()` and `baz()` 
+would be converted to Verilog, and function `qux()` is compiled using the RISC-V 
+compiler and executed by the processors.  
+
+The `AXI` box in the figure represents an interconnect that is automatically 
+instantiated in the project and the HLS modules are automatically connected to it. 
+This interconnect allows the communication between the processor and the HLS 
+modules.
+
+# The SmartHLS Processing Pipeline
+
+The actual set of functions that compose the SmartHLS pipeline in this example 
+and the effect of each transformation can be seen in the following figure:
+
+<img src="sev-kit-reference-design/doc/hls_pipeline.png" width="100%">
+
+The processing functions were chosen to range from simple, like the pixel inversion,
+to a bit more complex, like sobel and rotozoom.  The intention of this example is 
+to focus on the overall compilation flow and demonstrate a usage pattern rather 
+than achieving the best possible transformations. In the end, the idea is to 
+let developers replace these transformations with their own. 
+
+
+The design has two datapaths that the video frames can go through. One path is 
+through the DDR memory such that CPU can process the video frames, and the other 
+path is through the FPGA fabric. The video frames are steered via the `get_frame()` 
+and `put_frame()` C++ functions and controlled by the CPU.
+
+The second data path uses the FPGA fabric, the CPU can enable or disable each function 
+independently. When a specific function is disabled, then the video frame is just 
+forwarded to the next module.
+
+One additional transformation that only happens on the CPU path is that a small 
+Microchip logo bounces around the screen as it is overlaid on top of the 
+video frames from the camera. This is useful to show some activity in the background 
+when the frame rate slows down as the CPU has to perform additional transformations.
+
+Following is a brief description of the individual transformations:
+
+## get_pixel()
+
+This function can be thought of as a demux module. It receives a video stream 
+and either forwards it downstream or writes it to DDR memory depending on an option 
```

Comment:
Avoid "it" here to improve clarity. **This function** receives .... either forwards **the input** downstream or writes **the input** to DDR memory